File: torre.py
import peca

class Torre(peca.Peca):
   def __init__(self, posicao_origem_x, posicao_origem_y, cor):
       super().__init__(posicao_origem_x, posicao_origem_y, cor)



   def validar(self, posicao_x, posicao_y):
       if super().validar(posicao_x, posicao_y):
           if (posicao_x != self.posicao_atual_x ) and (posicao_y != self.posicao_atual_y):
             return False

           if((posicao_x== self.posicao_atual_x or posicao_y== self.posicao_atual_y) and
             (posicao_x!= self.posicao_atual_x or posicao_y!= self.posicao_atual_y)):

            return True


   def movimentar(self, destino_x, destino_y):

           if self.validar(destino_x, destino_y):
               self._posicao_atual_x= destino_x
               self._posicao_atual_y= destino_y
               return True
           return False

File: src/xadrez_jv/Torre.java
package xadrez_jv;

public class Torre extends Peca{
	
	Torre(int posicaoX, int posicaoY, int cor)
	{
		super(posicaoX, posicaoY, cor);
	}
	
	@Override 
	public boolean validar(int posicaoX, int posicaoY)
	{
		if (super.validar(posicaoX, posicaoY))
		{
			if (posicaoX != this.posicaoAtualX && posicaoY != this.posicaoAtualY)
				return false;
			if ((posicaoX == this.posicaoAtualX || posicaoY == this.posicaoAtualY)
				&& posicaoX != this.posicaoAtualX || posicaoY != this.posicaoAtualY)
				return true;
		}
		return true;
	}
	@Override 
	public boolean movimentar(int destinoX, int destinoY)
	{
		if (validar(destinoX, destinoY))
		{
			this.posicaoOrigemX = this.posicaoAtualX;
			this.posicaoOrigemY = this.posicaoAtualY;
			this.posicaoAtualX = destinoX;
			this.posicaoAtualY = destinoY;
			return true;
		}
		return false;
	}
}
